src/js/components/navbars.js

```diff
@@ -85,48 +86,127 @@
-   *  <div class="navbar navbar-app navbar-absolute-bottom">
-   *    <!-- ... -->
-   *  </div>
-   * ```
-   */
+  * @directive navController
+  * @restrict C
+  * @description
+  *
+  * Directive to bind controller to navbar
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-top' nav-controller>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
+  module.directive('navController', function() {
+    return {
+      restrict: 'A',
+      controller: function() {
+        this.registerController = function(name, ctrl) {
+          this[name] = ctrl;
+        };
+      },
+      link: function(scope) {}
+    };
+  });
+
+ /**
+  * @directive navbarAbsoluteTop
+  * @restrict C
+  * @description
+  *
+  * Setup absolute positioned top navbar.
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-top'>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
+
+ /**
+  * @directive navbarAbsoluteBottom
+  * @restrict C
+  * @description
+  *
+  * Setup absolute positioned bottom navbar.
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-bottom'>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
   angular.forEach(['top', 'bottom'], function(side) {
     var directiveName = 'navbarAbsolute' + side.charAt(0).toUpperCase() + side.slice(1);
     module.directive(directiveName, [
       '$rootElement',
       function($rootElement) {
         return {
           restrict: 'C',
-          link: function(scope) {
+          require: 'navController',
+          link: function(scope, elem, attr, ctrl) {
             $rootElement.addClass('has-navbar-' + side);
             scope.$on('$destroy', function() {
               $rootElement.removeClass('has-navbar-' + side);
             });
+            if (!global) global = ctrl;
+            global.registerController(side, new (function() {
+              this.hide = function() {
+                $rootElement.removeClass('has-navbar-' + side);
+                elem.addClass('ng-hide');
+              };
+              this.show = function() {
+                elem.removeClass('ng-hide');
+                $rootElement.addClass('has-navbar-' + side);
+              };
+              this.slideout = function() {
+                $rootElement.removeClass('has-navbar-' + side);
+                elem.addClass('ng-hide');
+              };
+              this.slidein = function() {
+                elem.removeClass('ng-hide');
+                $rootElement.addClass('has-navbar-' + side);
+              };
+            })());
           }
         };
       }
     ]);
   });
+
+  /**
+   * @directive [hide|show|slidein|slideout][Top|Bottom|Both][Nav]
+   * @restrict C
+   * @description
+   *
+   * Toggle each(or both) nabar(s) with or without animations
+   *
+   * ``` html
+   *  <view class='[hide|show|slidein|slideout]-[top|bottom|both]-[nav]'>
+   *    <!-- ... -->
+   *  </view>
+   * ```
+   */
+  angular.forEach(['Top', 'Bottom', 'Both'], function(nav) {
```

Comment:
``` 1) easily allow to hide, show the navbars (the slide part is unnecessary cause is better and easier achieved and customized via css) ```

I can agree with that.

``` Currently what you achieved with 2 directives could be done with a few css classes. Let's discuss outside the PR. ```

I don't think this is right though. The styles that fix the navbars are spread among classes at multiple node levels. I didn't really analyze the merit of that approach, because I assumed that there was a good reason for doing it(though I would have preferred if the height of the navbars could have been controlled by the font-size of the the nested copy). So you see in order to "eliminate" the styles at every level the controller has to know of all styles it is responsible for.

Now I did, at one time, use a hacky css override like, I am assuming, you suggest. But that, of course, made it unnecessarily cumbersome to switch between states when the view state changes.

This is much more versatile.